File: microservice-transaction-sample-with-shared-cluster-with-jdbc/client/src/main/java/sample/client/command/GetOrderCommand.java
package sample.client.command;

import io.grpc.ManagedChannel;
import java.util.concurrent.Callable;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;
import sample.rpc.GetOrderRequest;
import sample.rpc.GetOrderResponse;
import sample.rpc.OrderServiceGrpc;

@Command(name = "GetOrder", description = "Get order information by order ID")
public class GetOrderCommand implements Callable<Integer> {

  @Parameters(index = "0", paramLabel = "ORDER_ID", description = "order ID")
  private String orderId;

  @Override
  public Integer call() {
    ManagedChannel channel = Utils.getOrderServiceChannel();
    try {
      OrderServiceGrpc.OrderServiceBlockingStub stub = OrderServiceGrpc.newBlockingStub(channel);
      GetOrderResponse response =
          stub.getOrder(GetOrderRequest.newBuilder().setOrderId(orderId).build());
      Utils.printJsonString(response);
      return 0;
    } catch (Exception e) {
      e.printStackTrace();
      return 1;
    } finally {
      Utils.shutdownChannel(channel);
    }
  }
}

File: dotnet/microservice-transactions-sample-with-shared-cluster-with-linq/Client/Commands/GetOrderCommand.cs
using System.CommandLine;
using MicroserviceTransactionsSample.Client.Commands.Binders;
using MicroserviceTransactionsSample.Rpc;
using static MicroserviceTransactionsSample.Rpc.OrderService;

namespace MicroserviceTransactionsSample.Client.Commands;

public static class GetOrderCommand
{
    private const string Name = "GetOrder";
    private const string Description = "Get order information by order ID";

    private const string ArgName = "id";
    private const string ArgDescription = "order ID";

    public static Command Create()
    {
        var orderIdArg = new Argument<string>(ArgName, ArgDescription);
        var getOrderCommand = new Command(Name, Description)
        {
            orderIdArg
        };

        getOrderCommand.SetHandler(getOrder,
                                   orderIdArg,
                                   new OrderServiceClientBinder());
        return getOrderCommand;
    }

    private static async Task getOrder(string orderId, OrderServiceClient client)
    {
        var request = new GetOrderRequest { OrderId = orderId };
        var response = await client.GetOrderAsync(request);

        Console.WriteLine(response);
    }
}
